### DISCUSSION TITLE
Inconsistent output for a function call from postgrest API url

### DISCUSSION THREAD

**Discussion by alan-francis** [PostgREST/postgrest#1835] [Q&A] [answered]
Hi
while calling a function from my postgrest api url I am getting the result with the column name like below.

URL : http://localhost:16000/rpc/fn_name?terms=patents

Result:
```
[{
"fn_name":[
    1,
    [
      "Patents"
    ],
    null,
    [
      [
        "Patents: Patents"
      ]
    ]
  ]
}] 
```
But after I restart my postgrest docker my result changes to the one below without the function name.
```
[
  1,
  [
    "Patents"
  ],
  null,
  [
    [
      "Patents: Patents"
    ]
  ]
]
```
My desired out put is the one without the function name. But I don't know why this inconsistency in the output from postgrest. I tried sending the server a SIGUSR1 signal and also schema reloading.

**Comment by wolfgangwalther** [comment]
If restarting the server helps, but sending a signal or "reload the schema" does not - then you're doing something wrong on the latter. Restarting the server reloads the schema, too.

Which version of PostgREST are you using?

**Comment by alan-francis** [comment]
Thanks for the reply. The postgREST version we are using is  v7.0.1. First of all, I am not sure which one is the correct output. Once the postgrest is up and running, and I create a function and call it from the API, I am getting the result as an array of object with the function name as the object key. If I restart the postgrest docker, which should reload the schema as well, I am getting the result without the function name (which means the value of the previous results' object key). As i would be doing my db schema, view, function creation dynamically, the results are not consistent as it depends on whether my docker is restarted before the schema/view/function generation. I hope my issue is more clear now.

**Comment by wolfgangwalther** [comment, marked as answer]
> Thanks for the reply. The postgREST version we are using is v7.0.1. First of all, I am not sure which one is the correct output.

The output without the function name is correct. In v7.0.1 this didn't work for all functions, depending on the returned type. We have since cleaned this up a bit and the output should be more consistent in the latest nightly. However, I don't think this is something you are experiencing, because your output is correct once the schema cache is reloaded.

We need an uptodate schema cache to be able to tell which return type a function has and then properly format it. There is no way around that.

> Once the postgrest is up and running, and I create a function and call it from the API, I am getting the result as an array of object with the function name as the object key. If I restart the postgrest docker, which should reload the schema as well, I am getting the result without the function name (which means the value of the previous results' object key).

That's all expected.

> As i would be doing my db schema, view, function creation dynamically, the results are not consistent as it depends on whether my docker is restarted before the schema/view/function generation. I hope my issue is more clear now.

In our [latest nightly](https://github.com/PostgREST/postgrest/releases/tag/nightly) we have added the ability to reload the schema cache via postgres' `NOTIFY`. You can use this to create an event trigger to automatically reload the schema after a change. See here for an example: https://github.com/PostgREST/postgrest/issues/1512#issuecomment-643687504
